bi_sql_editor/models/bi_sql_view.py

```diff
@@ -0,0 +1,542 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from datetime import datetime
+from psycopg2 import ProgrammingError
+
+from openerp import _, api, fields, models, SUPERUSER_ID
+from openerp.exceptions import Warning as UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class BiSQLView(models.Model):
+    _name = 'bi.sql.view'
+    _inherit = ['sql.request.mixin']
+
+    _sql_prefix = 'x_bi_sql_view_'
+
+    _model_prefix = 'x_bi_sql_view.'
+
+    _sql_request_groups_relation = 'bi_sql_view_groups_rel'
+
+    _sql_request_users_relation = 'bi_sql_view_users_rel'
+
+    _STATE_SQL_EDITOR = [
+        ('model_valid', 'SQL View and Model Created'),
+        ('ui_valid', 'Graph, action and Menu Created'),
+    ]
+
+    technical_name = fields.Char(
+        string='Technical Name', required=True,
+        help="Suffix of the SQL view. (SQL full name will be computed and"
+        " prefixed by 'x_bi_sql_view_'. Should have correct"
```

Comment:
Syntax should follow: https://www.postgresql.org/"
        "docs/current/static/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS